### PR TITLE
Prefer atomic_inc to atomic_increment

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

**Review by dalg24 — CHANGES_REQUESTED** [review]
Per https://github.com/kokkos/kokkos/issues/7449#issuecomment-2414426030 `atomic_increment` might get deprecated in favor of `atomic_inc`

**Comment by aprokop** [comment]
> Per [kokkos/kokkos#7449 (comment)](https://github.com/kokkos/kokkos/issues/7449#issuecomment-2414426030) `atomic_increment` might get deprecated in favor of `atomic_inc`

Oh, sorry, I misunderstood, and thought it was the other way around.

File: src/cluster/detail/ArborX_FDBSCAN.hpp [hunk 1]
```diff
@@ -34,7 +34,7 @@ struct CountUpToN
   KOKKOS_FUNCTION auto operator()(Query const &query, Value const &) const
   {
     auto i = getData(query);
-    Kokkos::atomic_increment(&_counts(i));
+    Kokkos::atomic_inc(&_counts(i));
 
     if (_counts(i) < _n)
```
Comment (dalg24) [inline src/cluster/detail/ArborX_FDBSCAN.hpp:39]: This looks super sketchy
```suggestion
    auto const counts_i = Kokkos::atomic_inc_fetch(&_counts(i));

    if (counts_i < _n)
```

Comment (aprokop) [reply src/cluster/detail/ArborX_FDBSCAN.hpp:39]: It's not really sketchy, but you are right, it is could be suboptimal in the case of threads sharing a query (which they currently don't when running BVH). I'll fix it.

File: src/cluster/detail/ArborX_FDBSCANDenseBox.hpp [hunk 2]
```diff
@@ -74,16 +74,14 @@ struct CountUpToN_DenseBox
         int j = _permute(jj);
         if (distance(query_point, _primitives(j)) <= eps)
         {
-          Kokkos::atomic_increment(&count);
-          if (count >= _n)
+          if (Kokkos::atomic_inc_fetch(&count) >= _n)
```
Comment (dalg24) [inline src/cluster/detail/ArborX_FDBSCANDenseBox.hpp:77]: You don't want the value after the increment?

Comment (dalg24) [reply src/cluster/detail/ArborX_FDBSCANDenseBox.hpp:77]: Never mind

**Review by dalg24 — APPROVED** [review]

**Pull request merged**